vars/buildFailureMessage.groovy

```diff
@@ -0,0 +1,17 @@
+@NonCPS
+def call(){
+    def ERROR_STRING = "ERROR"
+    def message = ""
+    String delimiter = ","
+    def performance_log = currentBuild.rawBuild.getLog(Integer.MAX_VALUE).join('\n')
+    performance_log.eachLine() { line ->
+        if (line.matches(".*$ERROR_STRING.*")) {
+            message+=(line+delimiter)
+        }
+    }
+    message=message.replaceAll(",", "\n")
+    if(message.isEmpty()){
+        message="Build failed"
+    }
```

Comment:
Is this assuming the build exit with no `ERROR` keyword?

Reply:
Hey @peterzhuamazon yes you are right, this is for other outliers, like failed for any reason other than marked as ERROR, then in case instead we capture all of them, just sending a message as `Build failed`

Reply:
Got it.